### ISSUE TITLE
Java14 Full Records Support Check Validation: MagicNumberCheck

### ISSUE THREAD

**Issue by nmancus1** [checkstyle/checkstyle#8469]
Child of #8452 
Check documentation: https://checkstyle.sourceforge.io/config_coding.html#MagicNumber

From check documentation:
_Checks that there are no "magic numbers" where a magic number is a numeric literal that is not defined as a constant. By default, -1, 0, 1, and 2 are not considered to be magic numbers._

```
➜  full-record-grammar /usr/lib/jvm/java-14-openjdk/bin/javac --enable-preview --source 14 TestClass.java
Note: TestClass.java uses preview language features.
Note: Recompile with -Xlint:preview for details.
➜  full-record-grammar cat config.xml    
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
      <module name="MagicNumber"/>
  </module>
</module>
➜  full-record-grammar cat TestClass.java
public class TestClass {
    @anno(6) // violation
    public record MyRecord() {
        private static int myInt = 7; // violation

        public MyRecord{
            int i = myInt + 1; // no violation, 1 is defined as non-magic
            int j = myInt + 8; // violation
        }
        void foo() {
            int i = myInt + 1; // no violation, 1 is defined as non-magic
            int j = myInt + 8; // violation
        }

        public int hashCode() {
            return 10;    // violation
        }
    }
}
@interface anno {
    int value() default 10; // no violation
}
➜  full-record-grammar java $RUN_LOCALE -jar ~/IdeaProjects/checkstyle/target/checkstyle-8.35-SNAPSHOT-all.jar -c config.xml TestClass.java
Starting audit...
[ERROR] /home/nick/Desktop/full-record-grammar/TestClass.java:2:11: '6' is a magic number. [MagicNumber]
[ERROR] /home/nick/Desktop/full-record-grammar/TestClass.java:4:36: '7' is a magic number. [MagicNumber]
[ERROR] /home/nick/Desktop/full-record-grammar/TestClass.java:8:29: '8' is a magic number. [MagicNumber]
[ERROR] /home/nick/Desktop/full-record-grammar/TestClass.java:12:29: '8' is a magic number. [MagicNumber]
[ERROR] /home/nick/Desktop/full-record-grammar/TestClass.java:16:20: '10' is a magic number. [MagicNumber]
Audit done.
Checkstyle ends with 5 errors.
```
This check works as intended, I think we can close this issue.

**Comment by pbludov** [comment]
This check should be aware of `TokenTypes.RECORD_DEF` when the property `ignoreFieldDeclaration` is `true`:
```xml
<!DOCTYPE module PUBLIC "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
    "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="MagicNumber">
            <property name="ignoreFieldDeclaration" value="true"/>
        </module>
    </module>
</module>
```
```java
class TestClass {
    static int X = 42; // ok
}

record TestRecord() {
    static int X = 42; // unexpected violation
}
```
```
Starting audit...
[ERROR] /home/pbludov/src/tmp/TestClass.java:6:20: '42' is a magic number. [MagicNumber]
Audit done.
```

**Comment by strkkk** [comment]
fix is merged
